torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -946,6 +969,110 @@ void LoopNest::splitWithMask(For* f, int factor, For** outer, For** inner) {
   // TODO: record history of transformations
 }
 
+void LoopNest::reorderAxis(Tensor* t, For* a, For* b) {
+  if (a == b) {
+    // nothing to do.
+    return;
+  }
+  // find inner and outer.
+  For* outer{nullptr};
+  For* inner{nullptr};
+  std::deque<For*> internal_axes;
+
+  // Find relevant axes, store reversed.
+  for (For* loop : getLoopStmtsFor(t)) {
+    if (loop == a || loop == b) {
+      if (outer == nullptr) {
+        outer = loop;
+        internal_axes.push_front(loop);
+      } else {
+        inner = loop;
+        internal_axes.push_front(loop);
+      }
+    } else if (outer && !inner) {
+      internal_axes.push_front(loop);
+    }
+  }
+
+  if (!inner || !outer) {
+    throw std::runtime_error("Reordered a loop not in LoopNest");
+  }
+
+  Block* root = dynamic_cast<Block*>(outer->get_parent());
+  CHECK(root);
+
+  // Do a shallow copy of the inner blocks.
+  Block* body = new Block({});
+  for (auto* s : inner->body()->stmts()) {
+    inner->body()->remove_stmt(s);
+    body->append_stmt(s);
+  }
+
+  For* before{outer};
+  For* after{nullptr};
+  For* last = internal_axes.front();
+  Stmt* newInner = body;
+
+  // This is the major complexity in loop reordering: handling statments not in
+  // the straight line of the reorder. To handle this we partition the tree into
```

Comment:
Could you please add an example here? I think I know what you're referring to, but verifying this with an example would definitely help.

Reply:
I have unit tests which serve as examples, do you mean add examples to the comment?

Reply:
Yeah, I meant in the comment.